src/common/error.ts

```diff
@@ -0,0 +1,13 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { dialog } from "electron";
+
+export function error(title: string, err: string) {
+
+    dialog.showErrorBox("Internal Thorium ERROR", `${title}\nERROR: ${err}\n\nYou should probably restart the application`);
+}
```

Comment:
Great :)
Just for discussion:
It would be interesting to run a search in the codebase for all `await` function calls, and to identify how many of them do not have a `try+catch` handler for Promise rejection. I think there are many instances of unhandled rejections.

The problem is that developers assume that missing `try+catch` is fine because the exception simply gets cascaded up the call chain, and that at some point it will be handled. This is only partially true.

For better error handling practice, it is better to think of `try+catch` as the equivalent of `promise.then().catch()` ... which makes it much more obvious that promise rejections should be handled "locally" (using the contextual logic of the promise's intended purpose), rather than assuming that a higher-level / less-local logical layer of the code will somehow automagically handle the rejection appropriately.

What do you think?